rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -57,44 +59,79 @@ def replication_enabled(self):
         """Check whether replication enabled."""
         return ENVIRONMENT.get_value("REPLICATION_TO_RELATION_ENABLED", default=False, cast=bool)
 
-    def generate_relations_from_current_state_of_role(self):
+    def get_current_role_relations(self):
+        """Get current roles relations."""
+        return self.current_role_relations
+
+    def load_relations_from_current_state_of_role(self):
         """Generate relations from current state of role and UUIDs for v2 role and role binding from database."""
         if not self.replication_enabled():
             return
         try:
             logger.info(
                 "[Dual Write] Generate relations from current state of role(%s): '%s'", self.role.uuid, self.role.name
             )
-            relations = migrate_role(
-                self.role, False, str(self.root_workspace.uuid), self.org_id, True, True, True, False
+
+            self.binding_mapping = self.role.binding_mapping
+
+            if self.binding_mapping is None:
+                raise Exception(f"BindingMapping for role {self.role.uuid} not found.")
+
+            relations, _ = migrate_role(
+                self.role,
+                write_relationships=False,
+                root_workspace=str(self.root_workspace.uuid),
+                default_workspace=self.org_id,
+                current_bindings=self.binding_mapping,
```

Comment:
I changed this up a bit more than I intended for a few reasons:

- We were querying the binding mapping redundantly
- Now it's locked from the get-go, so it can be retrieved from the Role object
- Passing it around ended up cleaning up some of the parameters I found. Instead of controlling inner behavior with flags, I moved some of that behavior out here, and some of it is now controlled by passing the current bindings or not. The intent is just that it is maybe easier to use / harder to end up with inconsistent data now.